store/auth.go

```diff
@@ -225,3 +227,39 @@ func RefreshDischargeMacaroon(discharge string) (string, error) {
 
 	return requestDischargeMacaroon(UbuntuoneRefreshDischargeAPI, data)
 }
+
+// RequestStoreDeviceNonce requests a nonce for device authentication against the store.
+func RequestStoreDeviceNonce() (string, error) {
```

Comment:
The endpoint is for device identity, yes (https://myapps.developer.ubuntu.com/docs/api/device-auth.html).
I'm keeping here auth related methods hitting store endpoints (ie associated to the store implementation) because that's what I understood from the original code. I can refactor if we think there is a better approach.